pkg/serviceaccounttoken/secret.go

```diff
@@ -48,16 +48,17 @@ func EnsureSecretForServiceAccount(ctx context.Context, secretGetter secretGette
 		}
 		// k8s >=1.24 does not store a reference to the secret, but we need it to refer back to later
 		saCopy := sa.DeepCopy()
-		saCopy.Secrets = append(saCopy.Secrets, v1.ObjectReference{Name: secret.Name})
```

Comment:
Nit: 

The docstring for this field states the following:
>  This field should not be used to find auto-generated service account token secrets for use outside of pods.

This is very similar to what we are using this for (ours aren't auto-generated, but still). I was thinking it might be good to not rely on this field for this purpose, and instead to mark the templated secret with a label that we can use to look up the secret later on with a label selector. That being said, that would be a much more impactful change and I think it can be argued that this technically isn't against the docstring (since we manually made this secret), so marking this as a nit in case we want to stick more to the original issue.

Reply:
The same thing that prevents them from updating the SA and pointing to a different secret (i.e. RBAC). I'm kind of thinking that secret permissions would be a better RBAC check for this than service account permissions, since that more clearly aligns with user permissions to create a token for the SA (since that can be done through the token request api or by creating a secret with the appropriate annotation like we do).

Reply:
> If I knew or could read the label needed to link a secret to an SA, I could create that secret and thereby overtake the SA. Keeping it on the SA ensures that there is a 1:1 link between the secret and SA that Rancher created.

My point here is that the `secrets` field on the SA doesn't establish a 1-1 link. If I have `create` on secrets, I can create a secret with the `kubernetes.io/service-account.name` set, which will let me get a SA token for that service acocunt, regardless of if I have any permissions on that service account.

As far as I can see, this change would only change how data is injected into Rancher - in the current format I need 2 permissions - 1 on service accounts (to indicate what secret I want read into rancher) and 1 on secrets (to create the data that I want to feed into rancher), whereas in the new model I would only need 1 (secrets). However, I may be missing your point here, please correct me if you disagree?

Reply:
That is my point - in your proposal I would only need 1 permission, in the current model I need 2.

Reply:
Ok. I'll resolve the thread, I think it's hard to argue for something that would require less permissions.